projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-plugins-install-endpoint.php

```diff
@@ -90,6 +90,10 @@ protected function install() {
 			}
 		}
 
+		if ( ! $result ) {
+			return new WP_Error( 'plugin_install_failed', __( 'Plugin install failed because the result was invalid.', 'jetpack' ) );
```

Comment:
... This whole function is a WTF.

* When the bulk-install functionality was first added (7496bab33dda9dc3d8f7f041293f0d7d0ae1c130) the error handling was already confusing. When `$this->bulk` was false, there was always exactly one plugin so it didn't break when they put some error handling after the loop. And when it was true, no errors were handled (but `$this->log` was updated).
* Then 405b554a343f24a012b117a692f6262459deea80 weirdly put what became line 102 outside the loop as well, when it would have made more sense inside. Even then it had no `! $this->bulk` guard, but no warnings because an empty `$this->plugins` is rejected during validation so the loop would always run at least once. It also oddly changed the ordering of the non-bulk error handling.
* #7843 fixed a little bit of the weirdness in the misplaced after-the-loop error handling, which may have confused the intent going forward.
* #8464 refactored a bunch of stuff, and (unintentionally?) made it so bulk mode could now return an error but only if the last plugin failed.
* Now we reach this change. Which doesn't make things worse, it just continues the confusion.

A better fix would probably have been to just delete line 97 (no longer needed for non-bulk mode, and it running in bulk mode seems to be a bug) and move line 102 inside the loop (it being after was always a bug AFAICT).